README.md

```diff
@@ -3,45 +3,48 @@
 Let's roll your dices on Algorand!
 
 ## Overview
-[AlgoDices](https://testnet.algoexplorer.io/application/121287966) application
-let players **roll dices on-chain** (TestNet), thanks the [*Randomness Beacon Application*](https://developer.algorand.org/articles/usage-and-best-practices-for-randomness-beacon/).
+AlgoDices let players **roll dices on-chain** (both
+[TestNet](https://testnet.algoexplorer.io/application/149464577)
+and [MainNet]()), thanks the [*Randomness Beacon Application*](https://developer.algorand.org/articles/usage-and-best-practices-for-randomness-beacon/).
 
 Starting from AVM 7, Algorand enables [turstless randomness on chain](https://developer.algorand.org/articles/randomness-on-algorand),
 thanks to the use of [VRF](https://en.wikipedia.org/wiki/Verifiable_random_function):
 with the `vrf_verify` opcode an oracle Smart Contract can prove that a
 *pseudo-random value* has been honestly computed off-chain through a VRF process
 for a given blockchain round in the future.
 
-AlgoDices makes use of the Randomness Beacon App [110096026](https://testnet.algoexplorer.io/application/110096026)
-deployed on TestNet, which verifies, stores and provides the randomness to
+AlgoDices makes use of the Randomness Beacon App deployed on
+[TestNet](https://testnet.algoexplorer.io/application/110096026)
+and [MainNet](https://algoexplorer.io/application/947957720),
+which verifies, stores and provides the randomness to
 other applications for a given (future) round according to [ARC-21](https://arc.algorand.foundation/ARCs/arc-0021).
 
 ## ABI
 AlgoDices ABI exposes two methods:
 
-1. `book_die_roll`
-2. `roll_die`
+1. `book_dices_roll`
+2. `roll_dieces`
 
 AlgoDices App requires players to opt-in.
 
-### Book a (future) die roll
-Players can book a die roll for a future round, committing to future
+### Book (future) dices roll
+Players can book dices rolls for future round, committing to future
 randomness:
 
 ```json
 {
-    "name": "book_die_roll",
+    "name": "book_dices_roll",
```

Comment:
The App Call stores the future round in the player's Local State, so that they can use it once the randomness is available. How would you handle that without an explicit App Call?

Reply:
I think you can just specify the round you want at the moment you do the dice roll

Reply:
The advantage of `book_dices_roll` is that it makes it easier for the user of the smart contract to not make a mistake and to actually commit in advance to a round.
Also, it prevents the user from easily calling twice the smart contract from the same round, which may be an issue otherwise.

But the cost is quite high: one more transaction to make, that is not strictly required.

Reply:
I see. I think I will take the best of two worlds in a future PR, refactoring the methods as follows:

- Turning `dice_rolls` into an internal subroutine;
- Provide a `instant_roll`, which takes the round a player want as method input;
- Provide a `booked_roll`, which takes the round a player as previously committed to.